src/middlewared/middlewared/plugins/chart_releases_linux/portals.py

```diff
@@ -83,6 +86,27 @@ def get_ix_chart_portal(self, release_data, node_ip):
             portal_config['portalName']: [f'{portal_config["protocol"]}://{host}:{portal_config["port"]}']
         }
 
+
+    @private
+    def get_portals(self, release_data, node_ip):
+        portals = release_data['config'].get('portals')
+        
+        if not portals:
+            return {}
+        stored_portals = {}
+        for portal_config in portals:
+            if not portal_config.get('enabled'):
+                return {}
+            host = node_ip if portal_config['useNodeIP'] else portal_config['host']
+            if (portal_config['port'] == 80 AND portal_config['protocol'] == 'http') OR (portal_config['port'] == 443 AND portal_config['protocol'] ==  'https') OR (not portal_config['port']):
```

Comment:
Just a note but `AND` and `OR` aren't valid python operators.

Reply:
Yeah, I moved it away from draft too soon, because for some how github didn't show the CI.

Reply:
Cleaned all-of-it-up, thanks for the headsup!